[5.0.3 regression] Can’t `filter` a `QuerySet` on an `alias`

```py
from django.db import models

class Blog(models.Model):
    pass

class Entry(models.Model):
    blog = models.ForeignKey(Blog, on_delete=models.CASCADE)

def test() -> None:
    Blog.objects.alias(entries=models.Count("entry")).filter(entries__gt=5)
```

```
my_app/models.py:13: error: Cannot resolve keyword 'entries' into field. Choices are: entry, id  [misc]
Found 1 error in 1 file (checked 3 source files)
```

This was introduced by a28717df356fb767efb2d832b9146f759fd8cfd9 (#2277).

--- a/mypy_django_plugin/main.py
+++ b/mypy_django_plugin/main.py
@@ -169,6 +169,7 @@ class NewSemanalDjangoPlugin(Plugin):
             "values_list": partial(
                 querysets.extract_proper_type_queryset_values_list, django_context=self.django_context
             ),
+            "alias": partial(querysets.extract_proper_type_queryset_annotate, django_context=self.django_context),
             "annotate": partial(querysets.extract_proper_type_queryset_annotate, django_context=self.django_context),
             "create": partial(init_create.redefine_and_typecheck_model_create, django_context=self.django_context),
             "filter": typecheck_filtering_method,

--- a/tests/typecheck/managers/querysets/test_annotate.yml
+++ b/tests/typecheck/managers/querysets/test_annotate.yml
@@ -368,6 +368,7 @@
     qs.filter(annotate_wrong__gt=5)
 
     qs2 = qs.alias(alias_entries=models.Count("entry"))
+    qs2.filter(alias_entries__gt=5)
     qs2.filter(alias_wrong__gt=5, annotate_wrong__gt=5)
 
     Blog.objects.annotate().filter(xyz=1)
@@ -378,10 +379,10 @@
     )
   out: |
     main:6: error: Cannot resolve keyword 'annotate_wrong' into field. Choices are: entry, id, xyz  [misc]
-    main:9: error: Cannot resolve keyword 'alias_wrong' into field. Choices are: entry, id, xyz  [misc]
-    main:9: error: Cannot resolve keyword 'annotate_wrong' into field. Choices are: entry, id, xyz  [misc]
-    main:11: error: Cannot resolve keyword 'xyz' into field. Choices are: entry, id  [misc]
-    main:13: error: Cannot resolve keyword 'xyz' into field. Choices are: entry, id  [misc]
+    main:10: error: Cannot resolve keyword 'alias_wrong' into field. Choices are: entry, id, xyz, alias_entries  [misc]
+    main:10: error: Cannot resolve keyword 'annotate_wrong' into field. Choices are: entry, id, xyz, alias_entries  [misc]
+    main:12: error: Cannot resolve keyword 'xyz' into field. Choices are: entry, id  [misc]
+    main:14: error: Cannot resolve keyword 'xyz' into field. Choices are: entry, id  [misc]
   installed_apps:
     - myapp
   files:

